src/LCM2ROSControl.cpp

```diff
@@ -41,7 +41,18 @@ namespace valkyrie_translator
           return false;
         }
         handler_ = std::shared_ptr<LCM2ROSControl_LCMHandler>(new LCM2ROSControl_LCMHandler(*this));
-        
+
+        // Check which joints we have been assigned to
+        // If we have joints assigned to just us, claim those, otherwise claim all
+        std::vector<std::string> joint_names_;
+        if (!controller_nh.getParam("joints", joint_names_))
+          ROS_INFO_STREAM("Could not get assigned list of joints, will resume to claim all");
```

Comment:
Agreed. However for the purpose of a single controller only running on the robot without other dependencies this the preferred and recommended approach. The getParam is read from the yaml file where we specify whether to assign specific joints (e.g. the neck) or whether to grab all available joints (in the case of no joints specified in the parameter). LCM or director json config isn't compatible with ROS Control per se at the moment, but we can look into making it